jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileContextProviderTest.kt

```diff
@@ -0,0 +1,307 @@
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import com.intellij.testFramework.fixtures.JavaCodeInsightTestFixture
+import com.intellij.testFramework.runInEdtAndGet
+import com.intellij.testFramework.runInEdtAndWait
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererCsharp
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJava
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererKotlin
+import software.aws.toolkits.jetbrains.services.codewhisperer.util.DefaultCodeWhispererFileContextProvider
+import software.aws.toolkits.jetbrains.services.codewhisperer.util.FileContextProvider
+import software.aws.toolkits.jetbrains.utils.rules.HeavyJavaCodeInsightTestFixtureRule
+import software.aws.toolkits.jetbrains.utils.rules.addClass
+import software.aws.toolkits.jetbrains.utils.rules.addModule
+import software.aws.toolkits.jetbrains.utils.rules.addTestClass
+
+class CodeWhispererFileContextProviderTest {
+    @JvmField
+    @Rule
+    val projectRule = HeavyJavaCodeInsightTestFixtureRule()
+
+    lateinit var sut: DefaultCodeWhispererFileContextProvider
+
+    lateinit var fixture: JavaCodeInsightTestFixture
+    lateinit var project: Project
+
+    @Before
+    fun setup() {
+        fixture = projectRule.fixture
+        project = projectRule.project
+
+        sut = FileContextProvider.getInstance(project) as DefaultCodeWhispererFileContextProvider
+    }
+
+    @Test
+    fun `languages not supporting supplemental context will return empty`() {
+        val psiFiles = setupFixture(fixture)
+        val psi = psiFiles[0]
+
+        runBlocking {
+            var context = aFileContextInfo(CodeWhispererCsharp.INSTANCE)
+
+            assertThat(sut.extractSupplementalFileContextForSrc(psi, context)).isEmpty()
+            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isEmpty()
+
+            context = aFileContextInfo(CodeWhispererKotlin.INSTANCE)
+            assertThat(sut.extractSupplementalFileContextForSrc(psi, context)).isEmpty()
+            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isEmpty()
+
+            // TODO: the following assertions will fail as in test env, psiFile.virtualFile == null @psiGist.getFileData(psiFile)
+//            context = aFileContextInfo(CodeWhispererJava.INSTANCE)
+//            assertThat(sut.extractSupplementalFileContextForSrc(psi, context)).isNotEmpty
+//            assertThat(sut.extractSupplementalFileContextForTst(psi, context)).isNotEmpty
```

Comment:
nit: formatting

Reply:
let me reformat this, in fact I am thinkging to remove them